internal/output/diff.go

```diff
@@ -97,45 +97,96 @@ func ToDiff(out Root, opts Options) ([]byte, error) {
 		}
 
 		s += "\n\n"
+		s += "──────────────────────────────────\n"
+	}
+
+	hasDiffProjects := len(noDiffProjects)+len(erroredProjects) != len(out.Projects)
+
+	if hasDiffProjects {
+		s += fmt.Sprintf("Key: %s changed, %s added, %s removed",
+			opChar(UPDATED),
+			opChar(ADDED),
+			opChar(REMOVED),
+		)
+		s += "\n\n"
 	}
 
 	if len(erroredProjects) > 0 {
-		s += "──────────────────────────────────\n"
-		s += "\nThe following projects could not be evaluated: \n"
+		s += "The following projects could not be evaluated: \n"
 		for _, project := range erroredProjects {
 			s += project + "\n"
 		}
 		s += fmt.Sprintf("Run the following command to see more details: %s", ui.PrimaryString("infracost breakdown --path=/path/to/code"))
 		s += "\n\n"
 	}
 
-	if len(noDiffProjects) > 0 {
-		s += "──────────────────────────────────\n"
-		s += fmt.Sprintf("\nThe following projects have no cost estimate changes: %s", strings.Join(noDiffProjects, ", "))
-		s += fmt.Sprintf("\nRun the following command to see their breakdown: %s", ui.PrimaryString("infracost breakdown --path=/path/to/code"))
-		s += "\n\n"
+	if hasDiffProjects && out.DiffTotalMonthlyCost != nil && out.DiffTotalMonthlyCost.GreaterThan(decimal.Zero) {
+		s += tableForDiff(out)
+		s += "\n"
 	}
 
-	s += "──────────────────────────────────\n"
-	if len(noDiffProjects) != len(out.Projects) {
-		s += fmt.Sprintf("Key: %s changed, %s added, %s removed\n",
-			opChar(UPDATED),
-			opChar(ADDED),
-			opChar(REMOVED),
-		)
+	if len(noDiffProjects) > 0 {
+		if !hasDiffProjects && len(erroredProjects) > 0 {
+			s += "──────────────────────────────────\n"
+		}
+
+		if opts.ShowSkipped {
+			s += fmt.Sprintf("The following projects have no cost estimate changes: %s", strings.Join(noDiffProjects, ", "))
+			s += fmt.Sprintf("\nRun the following command to see their breakdown: %s", ui.PrimaryString("infracost breakdown --path=/path/to/code"))
+		} else {
+			s += fmt.Sprintf("%s had no cost estimate changes, rerun with --show-skipped to see details", english.Plural(len(noDiffProjects), "project", "projects"))
+		}
+
+		s += "\n\n"
+		s += "──────────────────────────────────"
 	}
 
 	unsupportedMsg := out.summaryMessage(opts.ShowSkipped)
 	if unsupportedMsg != "" {
-		if len(noDiffProjects) != len(out.Projects) {
-			s += "\n"
-		}
+		s += "\n"
 		s += unsupportedMsg
 	}
 
 	return []byte(s), nil
 }
 
+func tableForDiff(out Root) string {
+	t := table.NewWriter()
+	t.SetStyle(table.StyleBold)
+	t.Style().Format.Header = text.FormatDefault
+	t.AppendHeader(table.Row{
+		"Name",
```

Comment:
```suggestion
		"Project",
```
To keep it consistent with Infracost comment